rpcs3/util/cpu_stats.cpp

```diff
@@ -139,22 +151,21 @@ namespace utils
 			return;
 		}
 
-		PDH_FMT_COUNTERVALUE counterVal{};
 		DWORD dwBufferSize = 0; // Size of the pItems buffer
 		DWORD dwItemCount = 0;  // Number of items in the pItems buffer
-		PDH_FMT_COUNTERVALUE_ITEM *pItems = NULL; // Array of PDH_FMT_COUNTERVALUE_ITEM structures
+		PDH_FMT_COUNTERVALUE_ITEM *pItems = nullptr; // Array of PDH_FMT_COUNTERVALUE_ITEM structures
 
 		status = PdhGetFormattedCounterArray(m_cpu_cores, PDH_FMT_DOUBLE, &dwBufferSize, &dwItemCount, pItems);
 		if (PDH_MORE_DATA == status)
 		{
-			pItems = (PDH_FMT_COUNTERVALUE_ITEM*)malloc(dwBufferSize);
+			std::vector<u8> buffer(dwBufferSize);
+			pItems = reinterpret_cast<PDH_FMT_COUNTERVALUE_ITEM*>(buffer.data());
```

Comment:
Simply use vector<PDH_FMT_COUNTERVALUE_ITEM>, it is undefined behavior in c++ to tell the compiler that another object exists within an array by reinrepret_cast.